rdrf/rdrf/security_checks.py

```diff
@@ -0,0 +1,66 @@
+from registry.patients.models import Patient
+from registry.patients.models import ParentGuardian
+from django.core.exceptions import PermissionDenied
+import logging
+
+logger = logging.getLogger(__name__)
+
+def _get_prop(user, prop):
+    # case where site does not have prop defined
+    try:
+        return getattr(user, prop)
+    except:
+        return False
+
+def _user_is_patient_type(user):
+    return any([_get_prop(user, "is_patient"),
+                _get_prop(user, "is_parent"),
+                _get_prop(user, "is_carrier")])
+
+
+def _security_violation(user, patient_model):
+    logger.info("SECURITY VIOLATION User %s Patient %s" % (user.pk,
+                                                           patient_model.pk))
+    raise PermissionDenied()
+
+def security_check_user_patient(user, patient):
+    # either user is allowed to act on this record ( return True)
+    # or not ( raise PermissionDenied error)
+    if user.is_superuser:
+        return True
+
+    if patient is None:
+        return True
+
+    if type(patient) is int:
+        patient_model = Patient.objects.get(id=patient_id)
+    elif type(patient) is str:
+        patient_model = Patient.objects.get(id=int(patient_id))
+    else:
+        patient_model = patient
+    
+    if _user_is_patient_type(user):
```

Comment:
Perhaps remove on level of indentation (else can be redundant) and always raise at the end.

Reply:
nod - done